WalletWasabi.Fluent/ViewModels/Wallets/Send/LabelSelectionViewModel.cs

```diff
@@ -233,51 +235,43 @@ private void OnSelectionChanged()
 	{
 		_hiddenIncludedPockets.Clear();
 
-		var whiteListPockets =
-			NonPrivatePockets
-				.Where(pocket => pocket.Labels.All(pocketLabel => LabelsWhiteList.Any(labelViewModel => pocketLabel == labelViewModel.Value)));
+		var (isPrivateNeeded, isSemiPrivateNeeded) = IsPrivateAndSemiPrivatePocketNeeded();
 
-		if (IsPrivatePocketNeeded())
+		if (isPrivateNeeded)
 		{
 			_hiddenIncludedPockets.Add(_privatePocket);
 		}
-		else if (IsPrivateAndSemiPrivatePocketNeeded())
+
+		if (isSemiPrivateNeeded)
 		{
-			_hiddenIncludedPockets.Add(_privatePocket);
 			_hiddenIncludedPockets.Add(_semiPrivatePocket);
 		}
 
-		var totalSelected = whiteListPockets.EffectiveSumValue(_feeRate) + _hiddenIncludedPockets.EffectiveSumValue(_feeRate);
-
-		EnoughSelected = totalSelected >= _targetAmount;
+		EnoughSelected = GetUsedPockets().EffectiveSumValue(_feeRate) >= _targetAmount;
 
 		this.RaisePropertyChanged(nameof(LabelsWhiteList));
 		this.RaisePropertyChanged(nameof(LabelsBlackList));
 	}
 
-	private bool IsPrivatePocketNeeded()
+	private (bool, bool) IsPrivateAndSemiPrivatePocketNeeded()
```

Comment:
Maybe rename this to `IsPrivateAndSemiPrivatePocketsNeeded`.

Reply:
Done